modules/renter/proto/negotiate.go

```diff
@@ -172,26 +172,29 @@ func negotiateRevision(conn net.Conn, rev types.FileContractRevision, secretKey
 func newRevision(current types.FileContractRevision, cost types.Currency) types.FileContractRevision {
 	rev := current
 
-	// need to manually copy slice memory
-	rev.NewValidProofOutputs = make([]types.SiacoinOutput, 2)
-	rev.NewMissedProofOutputs = make([]types.SiacoinOutput, 3)
-	copy(rev.NewValidProofOutputs, current.NewValidProofOutputs)
-	copy(rev.NewMissedProofOutputs, current.NewMissedProofOutputs)
-
 	// move valid payout from renter to host
-	rev.NewValidProofOutputs[0].Value = current.NewValidProofOutputs[0].Value.Sub(cost)
-	rev.NewValidProofOutputs[1].Value = current.NewValidProofOutputs[1].Value.Add(cost)
-
+	rev.NewValidProofOutputs = newRevisionWithAdjustedPayout(current.NewValidProofOutputs, 2, 0, 1, cost)
```

Comment:
`newRevisionWithAdjustedPayout` is not a good name for the function, as all it really does is return adjusted payouts.

Second issue is with the inputs, they are all numeric. The function call is not readable unless you already know the function definition, as you have no idea what the numbers correspond to.

Finally, I would say that correctness is less obvious here. It needs to be obvious that money is moving from the renter to the host, and from the renter to the void. This code is actually a little confusing to me (the existing code, not the proposed code), because I thought that you were supposed to be moving money from the host to the void as well, but we can see here that we're not doing that.

--------

One definitely unfortunate thing here is that we didn't explain in the comments why we need to manually copy the slice memory over. I can't really remember the full details but I remember it was important. Maybe @lukechampine remembers? I think it's because slices are pointers, and we're using the underlying memory elsewhere in other parts of the execution.

Reply:
Reading this code at all does require understanding that in file contracts, we represent the renter money in output 0, the host money in output 1, and the void money in output 2

Reply:
That's probably worth documenting somewhere